### PR TITLE
Train and test scripts based on objects

### PR DESCRIPTION
**Pull request by rodrigoalmeida94**

### CODE REVIEW

File: landcover/models.py [hunk 1]
```diff
@@ -8,6 +9,8 @@
 from architectures import UNet, FC_DenseNet, FCN_Small
 from utils import load_nlcd_stats
 
+MIRRORED_STRATEGY = tf.distribute.MirroredStrategy()
```
Comment (markusuwe) [inline landcover/models.py]: What is this used for?

File: landcover/models.py [hunk 2]
```diff
@@ -148,36 +151,37 @@ def unet_from_segmentation_models(
...
+                optimizer=optimizer,
+                loss={
+                    "outputs_hr": hr_loss(),
+                    "outputs_sr": sr_loss(nlcd_class_weights, nlcd_means, nlcd_vars),
+                },
+                loss_weights={"outputs_hr": (39.0 / 40.0), "outputs_sr": (1.0 / 40.0)},
+                metrics={"outputs_hr": accuracy},
+            )
+        else:
+            print("Loss function not specified, model not compiled")
```
Comment (markusuwe) [inline landcover/models.py]: Can we use the logger instead?

File: landcover/test_model_landcover.py [hunk 3]
```diff
@@ -156,114 +73,255 @@ def do_args(arg_list, name):
...
+        fns = self.load_tiles()
+        model = self.load_model()
+
+        for i in range(len(fns)):
+            tic = float(time.time())
+            naip_fn = os.path.join(self.data_dir, fns[i][0])
+            lc_fn = os.path.join(self.data_dir, fns[i][1])
+            nlcd_fn = os.path.join(self.data_dir, fns[i][2])
+
+            print("Running model on %s\t%d/%d" % (naip_fn, i + 1, len(fns)))
```
Comment (markusuwe) [inline landcover/test_model_landcover.py]: use logger instead.

File: tests/test_test_model_landcover.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import sys
```
Comment (markusuwe) [inline tests/test_test_model_landcover.py]: File name is a bit confusing. Don't have a good suggestion though.

File: landcover/models.py (hunk 1 above)
Comment (rodrigoalmeida94) [reply landcover/models.py]: This is my attempt to replace the GPU selection they were using before - supposedly when you compile a model with this scope, all the GPU are used for training. Used this as guide https://www.tensorflow.org/guide/distributed_training

File: tests/test_test_model_landcover.py (hunk 4 above)
Comment (rodrigoalmeida94) [reply tests/test_test_model_landcover.py]: here it's a bit strange, but not sure what could be better - leaving as is for now.

File: landcover/models.py (hunk 2 above)
Comment (rodrigoalmeida94) [reply landcover/models.py]: Yup!

**Pull request merged**